File: Excel/607 Challenge.py
import pandas as pd

path = "607 Sort Odd Numbers First.xlsx"
input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=15)
test = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=15).rename(columns=lambda col: col.split('.')[0])

sorted_input = input.apply(lambda col: sorted(col, key=lambda x: (-(x % 2), x)))

print(sorted_input.equals(test)) # True

File: Excel/607 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/607 Sort Odd Numbers First.xlsx"
input = read_excel(path, range = "A2:C16")
test  = read_excel(path, range = "E2:G16")

sort_column = function(column) {
  col = tibble(column) %>%
    mutate(mod = if_else(column %% 2 == 0, 1, 0)) %>% 
    arrange(mod, column) %>%
    select(-mod) %>%
    pull()
  return(col)
}

result = input %>%
  mutate(across(everything(),sort_column)) %>%
  set_names(names(test))

all.equal(result, test, check.attributes = F)
#> [1] TRUE
